### PR TITLE
Aggregate rewards

### PR DESCRIPTION
**Pull request by 0xSero**
Made aggregator for RIC rewards
atm there's a bug where it won't aggregate, the ground work has all been set up in redux, though. should be a simply fix

### CODE REVIEW

**Comment by 0xSero** [comment]
@hrishibhattu @abhisheknaiidu 

This is 90% done, minor bug where the numbers won't add up and only last value is set. If you'd like to finish it you can claim the bounty on it,

**Review by mikeghen — CHANGES_REQUESTED** [review]
I think you're incrementing the rate every time I change app pages, the rate was correct, then I went to the active streams page, the rate went up, then I went to launchpad, went up again. What started as like 1100 turned into like 5000. Check you're not constantly adding more to the reward on page loads or something.

Also, do a quick styling on this and itll be good to go, just say "Rewards: X RIC/mo."

**Comment by 0xSero** [comment]
@mikeghen ready for review

**Review by mikeghen — CHANGES_REQUESTED** [review]
this is still an issue for me. visit the active streams page that will do it:
> I think you're incrementing the rate every time I change app pages, the rate was correct, then I went to the active streams page, the rate went up, then I went to launchpad, went up again. What started as like 1100 turned into like 5000. Check you're not constantly adding more to the reward on page loads or something.

**Comment by mikeghen** [comment]
Looks good but still increases when I go from Markets to Active Stream page, I have these open:
Screen Shot 2022-12-04 at 3 47 24 PM

**Review by mikeghen — CHANGES_REQUESTED** [review]
I get the following error, I think you editted CORS rules:
```
getCoingeckoPrices.ts:39 TypeError: Failed to fetch
    at getCoingeckoPrices.ts:30:9
    at d (regeneratorRuntime.js:86:17)
    at Generator._invoke (regeneratorRuntime.js:66:24)
    at Generator.next (regeneratorRuntime.js:117:21)
    at r (asyncToGenerator.js:3:20)
    at c (asyncToGenerator.js:25:9)
    at asyncToGenerator.js:32:7
    at new Promise (<anonymous>)
    at asyncToGenerator.js:21:12
    at dd (main.5f2b6c82.chunk.js:1:277565)
    ...
/#/invest/rex-market:1 Access to fetch at 'https://api.coingecko.com/api/v3/coins/markets?vs_currency=USD&ids=dai,usd-coin,weth,wrapped-bitcoin,matic-network,richochet,weth,usd-coin,wrapped-bitcoin' from origin 'https://ricochet-frontend-git-fork-seroxdesign-aggregat-b26cc6-mikeghen.vercel.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Review by mikeghen — CHANGES_REQUESTED** [review]
I tried to create a new stream but the input box doesnt accept my input
Screen Shot 2022-12-07 at 3 08 31 PM

File: src/containers/main/InvestContainer/InvestMarket.tsx [hunk 1]
```diff
@@ -92,6 +97,26 @@ export const InvestMarket: FC<InvestMarketProps> = ({ handleStart, handleStop })
 		).toFixed(toFixed);
 	}
 
+	const handleSetAggregatedRewards = (reward_amount: number) => {
+		setAggregatedRewards((aggregatedRewards) => [...aggregatedRewards, reward_amount]);
+	};
+
+	//This needs to be viewed by someone, issue is I need to clean up aggregatedRewardsArr after I do the math, but updating it retriggers useEffect and app gets stuck in loop
+
+	useEffect(() => {
```
Comment (0xSero) [inline src/containers/main/InvestContainer/InvestMarket.tsx:106]: This useEffect has a param of aggregatedRewards.

But in order to make sure it works properly after doing a state change I need to set aggregatedRewards to [0] which retriggers useEffect

File: src/containers/main/InvestContainer/InvestMarket.tsx [hunk 2]
```diff
@@ -92,6 +97,26 @@ export const InvestMarket: FC<InvestMarketProps> = ({ handleStart, handleStop })
...
+		let aggregated = 0;
+		aggregatedRewards.forEach((reward) => {
+			aggregated = aggregated + reward;
+		});
+		if (aggregatedRICRewards && +aggregatedRICRewards !== aggregated) {
+			dispatch(addReward(`${aggregated}`));
+			setAggregatedRewards([0]);
```
Comment (abhisheknaiidu) [inline src/containers/main/InvestContainer/InvestMarket.tsx:113]: you need to add `return` after this, otherwise it will go on a loop.

**Review by abhisheknaiidu — CHANGES_REQUESTED** [review]

**Comment by 0xSero** [comment]
PR has been merged into #831 will be closing this now

**Pull request closed without merging**